### ISSUE TITLE
Paste in IE11 is pasted as text only

### ISSUE THREAD

**Issue by jemunk** [fraywing/textAngular#1476]
How come is text copy and pasted in same textangular field in IE11/Edge, pasted as text only (without HTML)?
Code just ends up in this code block:
```
/* istanbul ignore next: Handle legacy IE paste */
if (!clipboardData && window.clipboardData && window.clipboardData.getData) {
	pastedContent = window.clipboardData.getData("Text");
	processpaste(pastedContent);
	e.stopPropagation();
	e.preventDefault();
	return false;
}
```

If I take a simple contenteditable element, copy/pasting is with HTML.

**Comment by pbassut** [comment]
Yeah, that's weird.
Probably we're trying to support old versions of IE with a workaround or patch some weird bug and now that workaround/patch is not needed anymore with edge.
Though while fixing this issue we need to make sure to still support other IE versions.

**Comment by sarbull** [comment]
As far as i searched IE11 doesn't have a handler on `.getData('text/html')` hence the issue..

But it is so weird because if i have a `<div contenteditable="true"></div>`, copy some text from a html page and when i paste it pastes with all the css needed in order to reproduce the initial element.

**Comment by jemunk** [comment]
I've found this thread, but I haven't dived much into it:
[http://stackoverflow.com/questions/2176861/javascript-get-clipboard-data-on-paste-event-cross-browser](http://stackoverflow.com/questions/2176861/javascript-get-clipboard-data-on-paste-event-cross-browser)

**Comment by sarbull** [comment]
By default the IE pastes as HTML in a `contenteditable` div, but you can't `getData` as HTML. I kinda sort it out by using this code [0] and having some sort of callback with a regex on the innerHTML in order to normalize the pasted html(with classes and inline css) resulting in a simple `<p><b>Lorem</b> ipsum <u>dolor</u></p>`

[0] - http://codebits.glennjones.net/editing/getclipboarddata.htm

**Comment by jemunk** [comment]
Aha! I just realized that the code block I mentioned in the header of this thread, is added in the latest commit to the taBind.js. I bet that it is a bug added in that commit. I will try reverting and testing tomorrow.

[https://github.com/textAngular/textAngular/commit/b9863c2487221c0149f834e40d4d80a2adb9f3ca](https://github.com/textAngular/textAngular/commit/b9863c2487221c0149f834e40d4d80a2adb9f3ca)

**Comment by sarbull** [comment]
Yeah, it looks to be from there 👍. In my situation i wanted to have a clean pasted content in the `contenteditable` so i made my own richtext(initially using textAngular but when i noticed that it doesn't maintain format in IE11(company's supported browser, #fml) i dropped using it).

A clean pasted content/html in the ng-model behind, be aware that when pasting from a browser copy or Microsoft Word the html contains all sorts of inline css and custom classes used in their rendering engines.

Maybe i can raise a PR where i can try and clean all the pasted inputs, but from the starts it can't be done on IE because of the `.getData('Text')` only option.

**Comment by jemunk** [comment]
I just tried reverting the [https://github.com/textAngular/textAngular/commit/b9863c2487221c0149f834e40d4d80a2adb9f3ca](https://github.com/textAngular/textAngular/commit/b9863c2487221c0149f834e40d4d80a2adb9f3ca) and it now supports pasting into IE11 with formatting.
So I don't understand the reason for that commit. Maybe for supporting some very old IE's... Who knows.

**Comment by sarbull** [comment]
It's sad for IE11 that it doesn't have a handler for other data types on paste in the `clipboardData.getData`, it only has `clipboardData.getData('Text')`.

**Comment by sarbull** [comment]
Be careful, if you copy from Word to  textAngular with black background it will paste the text with white background.

**Comment by sarbull** [comment]
This means that IE knows how to paste html but you don't have access on parsing it. It pastes with lots of classes and inline styles.

**Comment by sarbull** [comment]
@jemunk can this be closed? was it solved?

**Comment by jemunk** [comment]
Even though I made my own fix, I am in the believe that this is a bug. As long no one has made a public fix, I can't recommend closing this.

**Comment by amarnathnn** [comment]
This same issue occurs in the Chrome browser as well. when pasting some content from Microsoft word to textAngular, it loses header text color and other styles. 
Example: 
I have attached the original text as an image. 
![textAngular_OriginalWordText](https://user-images.githubusercontent.com/13251165/143598954-3818d596-ed7a-485c-b495-12b2e081940c.PNG)

the word content pasted as below in textAngular editor.

What would you do? 
	You will be aligned to our Human Resources vertical and will help us in the function that deals with people and people related areas such as recruiting, learning, training, onboarding, compensation and benefits, performance management, organization development and culture.

do we any fix in chrome browser ?

**Comment by Gayathri-Baskar** [comment]
Yeah, it worked but In IE while executing the line "window.clipboardData.getData("Text");" it ask for the permission to access the clipboard data. It is an other way can I get the pasted value in JS?

**Comment by amarnathnn** [comment]
> we are using textAngular Html editor. when i paste the content from word document with bulletins, they are not  formatted properly it the chrome browser itself..
